src/Traces.Web/Services/ApaleoOneService.cs

```diff
@@ -0,0 +1,64 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.JSInterop;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+using Optional;
+using Traces.Common.Constants;
+using Traces.Web.Models;
+
+namespace Traces.Web.Services
+{
+    public class ApaleoOneService : IApaleoOneService
+    {
+        private readonly IJSRuntime _jsRuntime;
+
+        private readonly JsonSerializerSettings _jsonSerializerSettings = new JsonSerializerSettings
+        {
+            ContractResolver = new CamelCasePropertyNamesContractResolver()
+        };
+
+        public ApaleoOneService(IJSRuntime runtime)
+        {
+            _jsRuntime = runtime;
+        }
+
+        public async Task<ResultModel<bool>> NavigateToReservation(TraceItemModel traceItemModel)
+        {
+            try
+            {
+                if (string.IsNullOrEmpty(traceItemModel.ReservationId) ||
+                    string.IsNullOrEmpty(traceItemModel.PropertyId))
+                {
+                    throw new ArgumentException(TextConstants.ApaleoOneMessageItemIncomplete);
```

Comment:
```suggestion
                    throw new BusinessValidationException(TextConstants.ApaleoOneMessageItemIncomplete);
```

Use `BusinessValidationException` instead. They are the exceptions that we are handling ourselves. An `ArgumentException` can occur for other things, and if they actually happen we want to be made aware of it in the error logs. :)